### ISSUE TITLE
Error: discarding overlapping duplicate mRNA feature in version 2.0

### ISSUE THREAD

**Issue by luciaq21** [gpertea/stringtie#272]
Hi,

I am getting this error _Error: discarding overlapping duplicate mRNA feature_ when I run stringtie. I know there is a similar issue posted here

 already, but that was for version 1.3.4 or something like that. I want to note that I am using version 2.0. So my guess is that this issue won't be solved by simply using another version. I am using a gff3 file, my command line is this:
stringtie -p 8 -G $Gtf -o ${name}.gtf -l $name ${name}.bam
I am running this on a cluster, in case that makes a difference.

Thanks,
Lucía

**Comment by gpertea** [comment]
It seems to me that the gene and mRNA entries have the same ID. That is not a valid GFF3 file. You can probably just remove the gene feature entries altogether, they are not needed for StringTie. e.g. by doing something like
`grep -vP '\tgene\t' old.gff > new.gff`
.. and then just using new.gff instead of the old.gff for the -G option.

I also advise against using version 2.0, it may have some issues that were fixed since then. Version 2.1.2 is the last one available at this moment.

**Comment by luciaq21** [comment]
I did what you said and used the new gff, but now I get this error message:
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!
[samopen] no @SQ lines in the header.
[sam_read1] missing header? Abort!

**Comment by gpertea** [comment]
That seems to be unrelated to the issue reported initially. There's a problem with the input .bam file perhaps ? The .bam file you are providing there ( ${name}.bam ) seems to not be correctly specified, perhaps it's not really a .bam file.

**Comment by luciaq21** [comment]
This is what I get when I do samtools view for the bam file:
A00266:159:HHWJLDSXX:1:2464:14479:33520 99      A01     4797    255     101M    =       4848    152     GTAAGAGGTTCGTGTTTCATCAATATACGGTGTGTTTCTATACATTGTGCTCTATCAATCTCCTTAACCTTGTCATCTGCACATCCTTCCCATAATCGCTT   FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFF   NH:i:1  HI:i:1  AS:i:200        NM:i:0  MD:Z:101
A00266:159:HHWJLDSXX:1:2166:5122:21637  355     A01     4809    0       10M730119N91M   =       735100  730390  TGTTTCATCATTTGTGCAAACATTATAATGATTGAAAATAATAAAGTACGTACGTATAACAAAAGTCTTCTTTTATCCTCCACTTAAAGACACTTCTCCAA   FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,FFFFFF   NH:i:5  HI:i:5  AS:i:187        NM:i:0 MD:Z:101
A00266:159:HHWJLDSXX:1:2464:14479:33520 147     A01     4848    255     101M    =       4797    -152    CTATCAATCTCCTTAACCTTGTCATCTGCACATCCTTCCCATAATCGCTTCTAGTCAATGTCCAAGCACTCCCACGGCCAGAAGTCATGCTCGTCCCTTTT   FFFFFF,FFFFFFFFFFFFFFFF:FFFFFFFFFFF,FFFFFFFFFFFFFFFFFFFFFFFFFFFF:FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF   NH:i:1  HI:i:1  AS:i:200        NM:i:0  MD:Z:101
So the bam file is fine, that cannot be the issue. Also, I ran it with other bam files and  it works so that is not what causing the error

**Comment by luciaq21** [comment]
Yeah, it worked in the end. Thanks!
Btw, at the end, is it normal for there to be a tmp_ file along with the gtf files?

**Comment by gpertea** [comment]
No, those should be removed if the run finished successfully. But those might have been left behind by the previous runs who failed in the same directory.
